crates/prover-alloy/src/lib.rs

```diff
@@ -114,6 +120,22 @@ impl Provider for AlloyProvider {
             .hash;
         Ok(hash)
     }
+
+    async fn get_block_number(
+        &self,
+        block_hash: alloy::primitives::B256,
+    ) -> Result<u64, anyhow::Error> {
+        let number = self
+            .client
+            .get_block_by_hash(block_hash, BlockTransactionsKind::Hashes)
+            .await
+            .map_err(|error| anyhow::anyhow!("Failed to get L1 block number: {:?}", error))?
+            .ok_or(anyhow::anyhow!("target block {block_hash} does not exist"))?
+            .header
+            .number
+            .as_u64();
+        Ok(number)
```

Comment:
`number` is already a `u64`

```suggestion
            .number;
        Ok(number)
```